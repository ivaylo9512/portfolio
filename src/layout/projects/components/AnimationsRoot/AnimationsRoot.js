import RestaurantAnimation from '@svgAnimations/RestaurantAnimation/RestaurantAnimation'
import ChatAnimation from '@svgAnimations/ChatAnimation/ChatAnimation'
import './AnimationsRoot.scss'
import ExtensionsAnimation from '@svgAnimations/ExtensionsAnimation/ExtensionsAnimation'
import TasksAnimation from '@svgAnimations/TasksAnimation/TasksAnimation'
import ProfileAnimation from '@svgAnimations/ProfileAnimation/ProfileAnimation'
import AnimationWrapper from '../AnimationWrapper/AnimationWrapper'
import { useSelector } from 'react-redux';
import { getVisited } from 'app/slices/animationsSlice'

const Animations = () => {
    const visited = useSelector(getVisited);

    return(
        <div data-testid='animations' className={`animations ${visited ? visited : ''}`}>
            <AnimationWrapper SvgAnimation={TasksAnimation} className='tasks'/>
            <AnimationWrapper SvgAnimation={ProfileAnimation} className='profile'/>
            <AnimationWrapper SvgAnimation={RestaurantAnimation} className='restaurant'/>
            <AnimationWrapper SvgAnimation={ExtensionsAnimation} className='extensions'/>
            <AnimationWrapper SvgAnimation={ChatAnimation} className='chat'/>
        </div>
    )
}

export default Animations