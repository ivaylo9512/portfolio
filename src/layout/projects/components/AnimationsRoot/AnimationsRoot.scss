.animations{
    position: relative;
    height: 100%;
    transition: transform 1.5s;

    > div{
        position: relative;
        transition: transform 1.5s;

        &:hover{
            z-index: 2;
        }

        .translate-wrapper{
            transition: transform 1.5s;
        }
    }
}

@media only screen and (orientation: landscape) {
    .animations{
        display: flex;
        align-items: center;
        height: 100%;
    
        &.tasks-visited .tasks-animation-wrapper,
        &.tasks-visited .profile-animation-wrapper {
            transform: translate(20%, 15%);
        }
    
        &.profile-visited .tasks-animation-wrapper,
        &.profile-visited .profile-animation-wrapper {
            transform: translate(-20%, 30%);
        }

        & > div {
            height: 34%;
        
            .translate-wrapper{
                height: 100%;
    
                svg{
                     height: 100%;
                }
            }
        }
    
        .tasks-animation-wrapper {
            margin-top: 10vh;
            left: 5vh;
            width: 50vw;
        }
    
        .restaurant-animation-wrapper {
            position: relative;
            width: 50vw;
            margin-top: 4vh;
            align-self: flex-end;
            margin-bottom: 12vh;
            height: 31%;
            left: 76vh;
            margin-left: 5vw;
        }
    
        .extensions-animation-wrapper {
            height: 40%;
            align-self: flex-start;
            margin-top: 7vh;
            width: 50vw;
            display: flex;
            justify-content: flex-end;
            left: 52vh;
        }
    
        .profile-animation-wrapper {
            align-self: start;
            margin-top: 14vh;
            height: 21%;
            width: 45vw;
            display: flex;
            justify-content: flex-end;
        }
    
        .chat-animation-wrapper {
            height: 45%;
            width: 100vw;
            display: flex;
            justify-content: flex-end;
            margin-left: -23vh;
            right: 6vw;
        }
    }

    .scroll.first-slide .animations{
        &.extensions-visited .extensions-animation-wrapper,
        &.extensions-visited .restaurant-animation-wrapper {
            transform: translate(-28%, 15%);
        }
    }
 
}

@media only screen and (orientation: portrait) {
    .animations{
        > div{
            margin-bottom: 18%;
    
            .translate-wrapper{
                width: 66%;
                display: inline-block;
            
                svg{
                    width: 100%;
                }
            }
        }
    
        .restaurant-animation-wrapper {
            > div{
                margin-left: 2%;
            }
        }
    
        .extensions-animation-wrapper {
            text-align: center;
    
            > div{
                margin-left: 9%;
            }
        }
    
        .chat-animation-wrapper {
            text-align: center;
    
            > div{
                width: 41%;
            }
        }
        
        .tasks-animation-wrapper {
            text-align: right;
    
            > div{
                margin-right: 15%;
                margin-top: 8%;
            }
        }

    }
}