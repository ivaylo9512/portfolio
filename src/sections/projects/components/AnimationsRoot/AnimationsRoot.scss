.animations{
    position: relative;
    height: 100%;
    transition: transform 1.5s;

    > div{
        transition: transform 1.5s;
        
        &:hover{
            transform: scale(1.35);
        }
    }

    .restaurant-animation-wrapper{
        width: 25%;
        bottom: 0;
        right: 20%;
        position: absolute;
    }

}