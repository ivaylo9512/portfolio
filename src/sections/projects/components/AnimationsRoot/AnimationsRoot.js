import RestaurantAnimation from "@svgAnimations/RestaurantAnimation/RestaurantAnimation"
import ChatAnimation from "@svgAnimations/ChatAnimation/ChatAnimation"
import "./AnimationsRoot.scss"
import ExtensionsAnimation from "@svgAnimations/ExtensionsAnimation/ExtensionsAnimation"
import TasksAnimation from "@svgAnimations/TasksAnimation/TasksAnimation"
import ProfileAnimation from "@svgAnimations/ProfileAnimation/ProfileAnimation"
import AnimationWrapper from "../AnimationWrapper/AnimationWrapper"

const Animations = () => {
    return(
        <div className="animations">
            <AnimationWrapper SvgAnimation={TasksAnimation} className="tasks"/>
            <AnimationWrapper SvgAnimation={RestaurantAnimation} className="restaurant"/>
            <AnimationWrapper SvgAnimation={ExtensionsAnimation} className="extensions"/>
            <AnimationWrapper SvgAnimation={ChatAnimation} className="chat"/>
            <AnimationWrapper SvgAnimation={ProfileAnimation} className="profile"/>
        </div>
    )
}

export default Animations