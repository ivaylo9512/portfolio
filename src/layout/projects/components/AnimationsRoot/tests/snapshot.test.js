import { shallow } from 'enzyme';
import AnimationsRoot from '../AnimationsRoot';
import * as Redux from 'react-redux';

describe('AnimationsRoot snapshot tests', () => {
    let selectorSpy;

    beforeEach(() => {
        selectorSpy = jest.spyOn(Redux, 'useSelector');
    })

    const createWrapper = (isVisited) => {
        selectorSpy.mockReturnValue(isVisited);

        return shallow(
            <AnimationsRoot />
        )
    }

    it('should match snapshot with visited', () => {
        const wrapper = createWrapper('tasks');

        expect(wrapper).toMatchSnapshot();
    })

    it('should match snapshot without visited', () => {
        const wrapper = createWrapper();

        expect(wrapper).toMatchSnapshot();
    })
})