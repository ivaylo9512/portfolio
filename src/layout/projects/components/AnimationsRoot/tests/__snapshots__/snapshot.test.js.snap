// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnimationsRoot snapshot tests should match snapshot with visited 1`] = `
<div
  className="animations tasks"
  data-testid="animations"
>
  <AnimationWrapper
    SvgAnimation={[Function]}
    className="tasks"
  />
  <AnimationWrapper
    SvgAnimation={[Function]}
    className="profile"
  />
  <AnimationWrapper
    SvgAnimation={[Function]}
    className="restaurant"
  />
  <AnimationWrapper
    SvgAnimation={[Function]}
    className="extensions"
  />
  <AnimationWrapper
    SvgAnimation={[Function]}
    className="chat"
  />
</div>
`;

exports[`AnimationsRoot snapshot tests should match snapshot without visited 1`] = `
<div
  className="animations "
  data-testid="animations"
>
  <AnimationWrapper
    SvgAnimation={[Function]}
    className="tasks"
  />
  <AnimationWrapper
    SvgAnimation={[Function]}
    className="profile"
  />
  <AnimationWrapper
    SvgAnimation={[Function]}
    className="restaurant"
  />
  <AnimationWrapper
    SvgAnimation={[Function]}
    className="extensions"
  />
  <AnimationWrapper
    SvgAnimation={[Function]}
    className="chat"
  />
</div>
`;
